pkg/projects/resource_project_user_test.go

```diff
@@ -0,0 +1,171 @@
+package projects
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+type ArtifactoryUser struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+	Admin    bool   `json:"admin"`
+}
+
+func createTestUser(t *testing.T, projectKey string, username string, email string) {
+	restyClient := getTestResty(t)
+
+	user := ArtifactoryUser{
+		Email:    email,
+		Password: "Password1",
+		Admin:    false,
+	}
+
+	_, err := restyClient.R().SetBody(user).Put(fmt.Sprintf("/artifactory/api/security/users/%s", username))
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func deleteTestUser(t *testing.T, projectKey string, username string) {
+	restyClient := getTestResty(t)
+
+	_, err := restyClient.R().Delete(fmt.Sprintf("/artifactory/api/security/users/%s", username))
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestAccProjectUser(t *testing.T) {
+	name := fmt.Sprintf("tftestprojects%s", randSeq(10))
+	resourceName := fmt.Sprintf("project_project.%s", name)
+	projectKey := strings.ToLower(randSeq(6))
+
+	username1 := "user1"
+	email1 := fmt.Sprintf("%s@tempurl.org", username1)
+	username2 := "user2"
+	email2 := fmt.Sprintf("%s@tempurl.org", username2)
+	developeRole := "developer"
+	contributorRole := "contributor"
+
+	params := map[string]interface{}{
+		"name":            name,
+		"project_key":     projectKey,
+		"username1":       username1,
+		"username2":       username2,
+		"developeRole":    developeRole,
+		"contributorRole": contributorRole,
+	}
+
+	initialConfig := executeTemplate("TestAccProjectUser", `
+		resource "project_project" "{{ .name }}" {
```

Comment:
This is gross - maybe call it something like `jfrog_project`

Reply:
We don't have to follow HCL guideline for resource naming 😆 I can revert it back to just `project`. It is gross.